File: src/main/java/ru/market/pricing_service/model/entity/PriceList.java
package ru.market.pricing_service.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PriceList {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer priceListId;

    @Column(length = 50)
    private String type;

    @Column(nullable = false)
    private LocalDate effectiveDate;

    @Column(nullable = false)
    private LocalDate endDate;
}

File: src/main/java/ru/market/pricing_service/model/entity/PriceList.kt
package ru.market.pricing_service.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.time.LocalDate

@Entity
data class PriceList(
    @Id @GeneratedValue val priceListId: Int,
    @Column(length = 50) val type: String,
    @Column val effectiveDate: LocalDate,
    @Column val endDate: LocalDate
)